src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -88,33 +92,47 @@ public new void RemoveMetadataReference(string referencePath)
             base.RemoveMetadataReference(referencePath);
         }
 
-        public void AddProjectReference(IProjectContext project, MetadataReferenceProperties properties)
+        public void AddProjectReference(IWorkspaceProjectContext project, MetadataReferenceProperties properties)
         {
+            var abstractProject = GetAbstractProject(project);
+
             // AbstractProject and ProjectTracker track project references using the project bin output path.
             // Setting the command line arguments should have already set the output file name and folder.
             // We fetch this output path to add the reference.
-            var referencedProject = this.ProjectTracker.GetProject(project.Id);
+            var referencedProject = this.ProjectTracker.GetProject(abstractProject.Id);
             var binPathOpt = referencedProject.TryGetBinOutputPath();
             if (!string.IsNullOrEmpty(binPathOpt))
             {
                 AddMetadataReferenceAndTryConvertingToProjectReferenceIfPossible(binPathOpt, properties);
             }
         }
 
-        public void RemoveProjectReference(IProjectContext project)
+        public void RemoveProjectReference(IWorkspaceProjectContext project)
+        {
+            var referencedProject = GetAbstractProject(project);
+
+            // AbstractProject and ProjectTracker track project references using the project bin output path.
+            // We fetch this output path to remove the reference.
```

Comment:
I don't get this comment. Why are we adding references to the metadata output instead of just adding project references directly? This conversion mechanism originally existed only so we could do references between C# and VB correctly.